lib/api/build.rs

```diff
@@ -1,178 +1,215 @@
+    let target_arch = match env::var("CARGO_CFG_TARGET_ARCH").unwrap().as_str() {
+        "x86" => "X86_32",
+        "x86_64" => "X86_64",
+        "arm" => "ARM",
+        "aarch64" => "AARCH64",
+        "mips" => "MIPS",
+        "powerpc" => "POWERPC",
+        "powerpc64" => "POWERPC64",
+        other => panic!("Unsupported CARGO_CFG_TARGET_ARCH: {}", other),
+    };
+
+    // Cleanup tmp data from prior builds
+    let wamr_dir = PathBuf::from(&crate_root).join("third_party/wamr");
+    let zip_dir = PathBuf::from(&crate_root)
+        .join("third_party")
+        .join(ZIP_NAME);
+    let _ = std::fs::remove_dir_all(&wamr_dir);
+    let _ = std::fs::remove_dir_all(&zip_dir);
+
+    // Fetch & extract wasm-micro-runtime source
+    let zip = ureq::get(WAMR_ZIP).call().expect("failed to download wamr");
+    let mut zip_data = Vec::new();
+    zip.into_reader()
+        .read_to_end(&mut zip_data)
+        .expect("failed to download wamr");
+    zip::read::ZipArchive::new(std::io::Cursor::new(zip_data))
+        .expect("failed to open wamr zip file")
+        .extract(&zip_dir)
+        .expect("failed to extract wamr zip file");
+    let _ = std::fs::remove_dir_all(&wamr_dir);
+    std::fs::rename(zip_dir.join(ZIP_NAME), &wamr_dir).expect("failed to rename wamr dir");
+
+    let wamr_platform_dir = wamr_dir.join("product-mini/platforms").join(target_os);
+    let mut dst = Config::new(wamr_platform_dir.as_path());
+
+    dst.always_configure(true)
+        .generator("Ninja")
+        .no_build_target(true)
+        .define(
+            "CMAKE_BUILD_TYPE",
+            if cfg!(debug_assertions) {
+                "RelWithDebInfo"
+            } else {
+                "Release"
+            },
+        )
+        .define("WAMR_BUILD_AOT", "0")
+        //.define("WAMR_BUILD_TAIL_CALL", "1")
+        //.define("WAMR_BUILD_DUMP_CALL_STACK", "1")
+        // .define("WAMR_BUILD_CUSTOM_NAME_SECTION", "1")
+        // .define("WAMR_BUILD_LOAD_CUSTOM_SECTION", "1")
+        .define("WAMR_BUILD_BULK_MEMORY", "1")
+        .define("WAMR_BUILD_REF_TYPES", "1")
+        .define("WAMR_BUILD_SIMD", "1")
+        .define("WAMR_BUILD_FAST_INTERP", "1")
+        .define("WAMR_BUILD_LIB_PTHREAD", "1")
+        .define("WAMR_BUILD_LIB_WASI_THREADS", "0")
+        .define("WAMR_BUILD_LIBC_WASI", "0")
+        .define("WAMR_BUILD_LIBC_BUILTIN", "0")
+        .define("WAMR_BUILD_SHARED_MEMORY", "1")
+        .define("WAMR_BUILD_MULTI_MODULE", "0")
+        .define("WAMR_DISABLE_HW_BOUND_CHECK", "1")
+        .define("WAMR_BUILD_TARGET", target_arch);
+
+    if target_os == "windows" {
+        dst.define("CMAKE_CXX_COMPILER", "cl.exe");
+        dst.define("CMAKE_C_COMPILER", "cl.exe");
+        dst.define("CMAKE_LINKER_TYPE", "MSVC");
+        dst.define("WAMR_BUILD_PLATFORM", "windows");
+        dst.define("WAMR_BUILD_LIBC_UVWASI", "0");
+    }
+
+    //if target_os == "ios" {
```

Comment:
So no iOS support for WAMR?

I don't think merging with WAMR only supported on some architectures is good... that create a feature matrix mess.
At the very least this should mention why it's commented out.

Reply:
(I'll reply to the other comments soon)

WAMR does support iOS. This bit was a hack to remove the `-mfloat-abi=` parameter from the make file in WAMR itself. Should not be needed anymore after https://github.com/bytecodealliance/wasm-micro-runtime/pull/3889 got merged.